ksqldb-engine/src/test/java/io/confluent/ksql/function/udf/array/ArrayJoinTest.java

```diff
@@ -0,0 +1,258 @@
+            .build()
+        )
+        .field("f5", SchemaBuilder.array(Schema.STRING_SCHEMA).build())
+        .field("f6", SchemaBuilder.array(SchemaBuilder.struct()
+            .field("k", Schema.STRING_SCHEMA)
+            .field("v", Schema.INT32_SCHEMA)
+            .build())
+        )
+        .field("f7",
+            SchemaBuilder.array(SchemaBuilder.array(SchemaBuilder.array(Schema.INT32_SCHEMA))))
+        .field("f8", SchemaBuilder.map(Schema.STRING_SCHEMA, Schema.INT32_SCHEMA).build())
+        .field("f9", SchemaBuilder
+            .map(Schema.STRING_SCHEMA, SchemaBuilder.array(Schema.STRING_SCHEMA).build()).build())
+        .field("f10", SchemaBuilder
+            .map(Schema.STRING_SCHEMA, SchemaBuilder.array(Schema.INT32_SCHEMA).build()).build())
+        .build();
+
+    STRUCT_DATA = new Struct(structSchema)
+        .put("f1", "ksqldb UDF sample data")
+        .put("f2", 42)
+        .put("f3", true)
+        .put("f4", new Struct(structSchema.field("f4").schema())
+            .put("f4-1", "hello ksqldb")
+        )
+        .put("f5", Arrays.asList("str_1", "str_2", "...", "str_N"))
+        .put("f6", Arrays.asList(
+            new Struct(structSchema.field("f6").schema().valueSchema())
+                .put("k", "a").put("v", 1),
+            new Struct(structSchema.field("f6").schema().valueSchema())
+                .put("k", "b").put("v", 2),
+            new Struct(structSchema.field("f6").schema().valueSchema())
+                .put("k", "c").put("v", 3)
+            )
+        )
+        .put("f7", Arrays.asList(
+            Arrays.asList(Arrays.asList(0,1), Arrays.asList(2,3,4), Arrays.asList(5, 6),
+                Arrays.asList(7,8,9)),
+            Arrays.asList(Arrays.asList(9,8,7),Arrays.asList(6,5), Arrays.asList(4,3,2),
+                Arrays.asList(1,0)
+            )
+        ))
+        .put("f8", new LinkedHashMap<String, Integer>() {{
+          put("k1", 6);
+          put("k2", 5);
+          put("k3", 4);
+        }})
+        .put("f9", new LinkedHashMap<String, List<String>>() {{
+          put("k1", Arrays.asList("v1-a", "v1-b"));
+          put("k2", Arrays.asList("v2-a","v2-b", "v2-c", "v2-d"));
+          put("k3", Arrays.asList("v3-a","v3-b","v3-c"));
+        }})
+        .put("f10", new LinkedHashMap<String, List<Integer>>() {{
+          put("k1", Arrays.asList(12, 21));
+          put("k2", Arrays.asList(23, 32));
+          put("k3", Arrays.asList(24, 42));
+        }});
+  }
+
+  @Test
+  public void shouldReturnNullForNullInput() {
+    assertNull(arrayJoinUDF.join(null));
+    assertNull(arrayJoinUDF.join(null,CUSTOM_DELIMITER));
+  }
+
+  @Test
+  public void shouldReturnEmptyStringForEmptyArrays() {
+    assertTrue(arrayJoinUDF.join(Collections.emptyList()).isEmpty());
+    assertTrue(arrayJoinUDF.join(Collections.emptyList(),CUSTOM_DELIMITER).isEmpty());
+  }
+
+  @Test
+  public void shouldReturnCorrectStringForFlatArraysWithPrimitiveTypes() {
```

Comment:
can you add decimal and long types into this? (see not above that `BigInteger` and `Long` are not the same in java)

Reply:
sure. but both or either of them? see my question above at the `BigInteger` comment.

Reply:
done!